src/main/java/underlay/rmi/RMIUnderlay.java

```diff
@@ -36,8 +40,16 @@ public RMIUnderlay(int port) {
       initRMI();
       host = new JavaRMIHost(this);
       LocateRegistry.createRegistry(port).rebind("RMIImpl", host);
-    } catch (Exception e) {
-      System.err.println("[RMIUnderlay] Error while initializing at port " + port);
+    } catch (ExportException ee) {
+      try {
+        System.out.println("error during creation");
+        LocateRegistry.getRegistry(port).rebind("RMIImpl", host);
+      } catch (RemoteException re) {
+        System.err.println("[RMIUnderlay] Error while getting registry at port " + port);
+        re.printStackTrace();
+      }
+    } catch (RemoteException e){
+      System.err.println("[RMIUnderlay] Error while creating registry at port " + port);
```

Comment:
Is this necessary?

Reply:
I added this to fix the `java.rmi.server.ExportException: internal error: ObjID already in use` error. More on that here: https://stackoverflow.com/questions/14982760/is-it-possible-to-manually-check-locateregistry-existing